contrib/postgresql/collectd_insert2.sql

```diff
@@ -0,0 +1,343 @@
+-- collectd - contrib/postgresql/collectd_insert2.sql
+-- Copyright (C) 2012 Sebastian 'tokkee' Harl
+-- Copyright (C) 2023 Georg Gast
+-- All rights reserved.
+--
+-- Redistribution and use in source and binary forms, with or without
+-- modification, are permitted provided that the following conditions
+-- are met:
+--
+-- - Redistributions of source code must retain the above copyright
+--   notice, this list of conditions and the following disclaimer.
+--
+-- - Redistributions in binary form must reproduce the above copyright
+--   notice, this list of conditions and the following disclaimer in the
+--   documentation and/or other materials provided with the distribution.
+--
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+-- AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+-- IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+-- ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+-- LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+-- CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+-- SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+-- INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+-- CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+-- ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+-- POSSIBILITY OF SUCH DAMAGE.
+
+-- Description:
+-- ------------
+--
+-- My development was done on postgresql 15.
+--
+-- This is a similar database as the one from Sebastian 'tokkee' Harl but it
+-- has some fundamental changes. It does not store all values in one big table
+-- that has a lot unneccessary stuff inside like millons of repeated "value"
+-- names.
```

Comment:
For something that "has fundamental changes", you should state those changes more clearly than just:
> It does not store all values in one big table with repeated "value" names.

E.g. something like:
> Instead of storing all values to a single "values" table with repeated "value" names, it ....

Where you describe at least the per-plugin functionality that your SQL seems to have [1], instead of it being more generic like the original one: https://github.com/collectd/collectd/blob/main/contrib/postgresql/collectd_insert.sql

[1] I've never used psql, and last time I used SQL, was in previous century, so somebody else needs to review the actual SQL code.